### DISCUSSION TITLE
<Link/> does not re-trigger getStaticProps

### DISCUSSION THREAD

**Discussion by ErnestBrandi** [vercel/next.js#17475] [Help] [answered]
Hi,
I am using the `Link` component in order to navigate through pages in my website.
On a certain page I use `getStaticPaths` and `getStaticProps` with `fallback: true` and `revalidate` options.
I noticed that, when a page has been built, querying it again through Link does not trigger `getStaticProps` again even if the `revalidate` time has been overpassed. 
If I reload the page (F5 or so), then I do have an newer version of the page.
Is this a normal behaviour ? 
Thanks

**Comment by ijjk** [comment, marked as answer]
Hi, this is the expected behavior for a client transition as data loaded for a page with `getStaticProps` is cached on the client once it is loaded to provide as fast page transitions as possible. Providing a mechanism to bypass this caching for client transitions may be investigated more if it seems beneficial for applications

**Reply by ErnestBrandi** [reply]
Ok I understand better now. 
My use case is the following :
- I populate my dynamic web page with data coming from a bunch of API (dynamic meaning it follows that format : `/path/[id]`)
- Some of those have a very restrictive quota (like 30 requests a minute)
- Because of this, I found the use of `getStaticProps` very appropriate as it could save me some API calls once a page is "built"
- But sometime a page is build when the limit as been reached and then it shows an error message "forever" (well until it's refreshed)

I would **love** a way to bypass that cache ! Ideally it would be an option to return from `getStaticProps`, as this would be (in my case) the moment I now something went wrong.
Also what would you recommand for now ? Fetching data client side (using `swr`) if I have no (or corrupt) data ?

**Reply by ErnestBrandi** [reply]
Also could you confirm (or not) the following :
- I have another page listing entries (this where the `<Link>` are), all leading to the dynamic page I talked before (so you can open `/path/foo`, `/path/bar`, etc).
- A few of those pages are being browsed (and therefore built)
- If someone comes back to that listing page and because of the `prefetch` behaviour of `Link`, the "json file" previously generated will be downloaded for all those links right ?
- So if the `revalidate` option is set to true, and because of the `stale-while-revalidate` behaviour of `getStaticProps`, will it re-execute its code in the background for each of those links ? (that would be bad concerning those quotas...)

**Reply by ijjk** [reply]
If the data is corrupt you shouldn't return it from `getStaticProps` currently and instead should throw an error so that it isn't cached. You can handle this client-side instead as you mentioned which will give you more control over how you handle the response. 

For the pre-fetching triggering revalidation questions, yes these can trigger revalidation if it is past the defined revalidation period. To prevent hitting the limits you can either increase the revalidation period or disable prefetching for those specific links as mentioned [in the docs](https://nextjs.org/docs/api-reference/next/link).

**Reply by ErnestBrandi** [reply]
So I can just throw an error (without catching it ?) in `getStaticProps` and nothing will be set in cache ? Shouldn't this be documented somewhere ? I am aware that I can disable prefetching, I just realized it was because of this the API quota's was reached.
Thanks for those answers !

**Reply by ijjk** [reply]
Correct, if you let the error be thrown in `getStaticProps` it won't be cached. I mentioned this to the team to see if we can get this added to our documentation.

**Reply by cbdeveloper** [reply]
I just ran into this caveat. I'm using `getStaticProps + revalidate` and I had to stop using `Link from next/link` because the user could be stuck on stale pages forever. If all my pages were revalidated during a user session, the user would still see stale pages because of the prefetched / client-cached json props for all the pages as they were when my user started his session.

We need a way to trigger page revalidation using `Link from next/link` otherwise it won't be possible to use client side transitions on pages that contain any dynamic data.
